src/Host/Client/Impl/rtvs/R/overrides.R

```diff
@@ -55,8 +55,9 @@ show_file <- function(files, header, title, delete.file) {
 }
 
 install.packages <- function(...) {
+    invisible(rtvs:::send_notification('!BeforePackagesInstalled'))
```

Comment:
That C# code is awaiting doesn't have any effect on R code here - `send_notification` sends the message and immediately returns. So all that await in C# code does currently is prevent further handling of new incoming messages until that notification is handled. For R to wait until C# callback completes, it needs to be changed to request/response.